gstd/src/common/handlers.rs

```diff
@@ -25,43 +25,45 @@
 //! debug and non-debug mode, for programs built in `wasm32` architecture.
 //! For `debug` mode it provides more extensive logging.
 
-#[cfg(target_arch = "wasm32")]
-use {crate::ext, core::alloc::Layout, core::panic::PanicInfo};
-
-#[cfg(target_arch = "wasm32")]
+#[cfg(feature = "oom-handler")]
 #[alloc_error_handler]
-pub fn oom(_: Layout) -> ! {
-    ext::oom_panic()
+pub fn oom(_: core::alloc::Layout) -> ! {
+    crate::ext::oom_panic()
 }
 
-#[cfg(not(any(feature = "debug", debug_assertions)))]
-#[cfg(target_arch = "wasm32")]
-#[panic_handler]
-pub fn panic(_: &PanicInfo) -> ! {
-    ext::panic("no info")
-}
+#[cfg(feature = "panic-handler")]
+mod panic_handler {
+    use crate::ext;
+    use core::panic::PanicInfo;
 
-#[cfg(any(feature = "debug", debug_assertions))]
-#[cfg(target_arch = "wasm32")]
-#[panic_handler]
-pub fn panic(panic_info: &PanicInfo) -> ! {
-    use crate::prelude::format;
+    #[cfg(not(all(feature = "panic-messages", feature = "debug")))]
+    #[panic_handler]
+    pub fn panic(_: &PanicInfo) -> ! {
+        ext::panic("no info")
+    }
 
-    let msg = match (panic_info.message(), panic_info.location()) {
-        (Some(msg), Some(loc)) => format!(
-            "'{:?}', {}:{}:{}",
-            msg,
-            loc.file(),
-            loc.line(),
-            loc.column()
-        ),
-        (Some(msg), None) => format!("'{msg:?}'"),
-        (None, Some(loc)) => {
-            format!("{}:{}:{}", loc.file(), loc.line(), loc.column())
-        }
-        _ => ext::panic("no info"),
-    };
+    #[cfg(all(feature = "panic-messages", feature = "debug"))]
+    #[panic_handler]
+    pub fn panic(panic_info: &PanicInfo) -> ! {
+        use crate::prelude::format;
```

Comment:
```suggestion
        use alloc::format;
```

And move to other imports

Reply:
Pretty sure it was `crate::prelude::format` prior to this PR; why should that be changed?

> And move to other imports

By "other imports" you mean https://github.com/gear-tech/gear/blob/8dc668a38d52843aaea01b5a8158d4bff613998c/gstd/src/common/handlers.rs#L36-L37?
It is only used in this implementation, the other one does not need it

Reply:
Became irrelevant with 5d9baa86f